server/server.go

```diff
@@ -435,7 +436,14 @@ func (a *ArgoCDServer) watchSettings() {
 			log.Infof("dex config modified. restarting")
 			break
 		}
-		if prevOIDCConfig != a.settings.OIDCConfigRAW {
+		newOIDCConfig := a.settings.OIDCConfig()
+		if prevOIDCConfig != nil && newOIDCConfig != nil {
+			if !reflect.DeepEqual(*prevOIDCConfig, *newOIDCConfig) {
+				log.Infof("oidc config modified. restarting")
+				break
+			}
+		}
+		if (prevOIDCConfig != nil && newOIDCConfig == nil) || (prevOIDCConfig == nil && newOIDCConfig != nil) {
```

Comment:
I didn't understand the problem description in your PR but it seems that this is the real fix you are addressing. Can you please confirm?

Reply:
No, I described above that currently the raw `non-interpolated` config is being compared but we should compare the interpolated one.